src/py/flwr/cli/build.py

```diff
@@ -0,0 +1,156 @@
+        Optional[Path],
+        typer.Option(help="The Flower project directory to bundle into a FAB"),
+    ] = None,
+    signed: Annotated[bool, typer.Option(help="Flag to sign the FAB")] = False,
+) -> None:
+    """Build a Flower project into a FAB file.
+
+    It can be ran without any argument:
+
+        `flwr build`
+
+    In which case, it will try to bundle the current directory if it is a
+    valid Flower App.
+
+    It can also be ran with a directory argument:
+
+        `flwr build --directory ./docs/target_dir`
+
+    And finally, the `--signed` argument can also be used to sign the FAB:
+
+        `flwr build --signed`
+
+    Note that this will prompt the user for its private key.
+    """
+    if directory is None:
+        directory = Path.cwd()
+
+    directory = directory.resolve()
+    if not directory.is_dir():
+        typer.secho(
+            f"❌ The path {directory} is not a valid directory.",
+            fg=typer.colors.RED,
+            bold=True,
+        )
+        raise typer.Exit(code=1)
+
+    if not is_valid_project_name(directory.name):
+        typer.secho(
+            f"❌ The project name {directory.name} is invalid, "
+            "a valid project name must start with a letter or an underscore, "
+            "and can only contain letters, digits, and underscores.",
+            fg=typer.colors.RED,
+            bold=True,
+        )
+        raise typer.Exit(code=1)
+
+    if validate_project_dir(directory) is None:
+        raise typer.Exit(code=1)
+
+    # Load .gitignore rules if present
+    ignore_spec = _load_gitignore(directory)
+
+    # Set the name of the zip file
+    fab_filename = f"{directory.name}.fab"
+    list_file_content = ""
+
+    allowed_extensions = {".py", ".toml", ".md"}
+
+    with zipfile.ZipFile(fab_filename, "w", zipfile.ZIP_DEFLATED) as fab_file:
+        for root, _, files in os.walk(directory, topdown=True):
+            # Filter directories and files based on .gitignore
+            files = [
+                f
+                for f in files
+                if not ignore_spec.match_file(Path(root) / f)
+                and f != fab_filename
+                and Path(f).suffix in allowed_extensions
+            ]
+
+            for file in files:
+                file_path = Path(root) / file
+                archive_path = file_path.relative_to(directory)
+                fab_file.write(file_path, archive_path)
+
+                # Calculate file info
+                sha256_hash = _get_sha256_hash(file_path)
+                file_size_bits = os.path.getsize(file_path) * 8  # size in bits
+                list_file_content += f"{archive_path},{sha256_hash},{file_size_bits}\n"
+
+        # Add LIST and LIST.jwt to the zip file
+        fab_file.writestr(".info/LIST", list_file_content)
+
+        # Optionally sign the LIST file
+        if signed:
+            secret_key = typer.prompt(
+                "Enter the secret key to sign the LIST file", hide_input=True
```

Comment:
```suggestion
                "Enter the path to the secret key to sign the LIST file", hide_input=True
```

Reply:
I think the input is the key, not the path, I will check. Should I modify the function to extract the key from a path?

Reply:
Isn't the common approach to pass a file path? With Android, you pass the path to the key store file: https://developer.android.com/build/building-cmdline#sign_manually